src/js/components/NameChoiceWindow.js

```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+import { saveName } from '../actions/nextStepCharAction';
+import { connect } from "react-redux"
+import "../../styles/attributes.scss";
+
+@connect((store) => {
```

Comment:
I m not qute sure why you always use `this.props.dispatch`
If you are using connect you can `mapDispatchToProps`

https://github.com/reactjs/react-redux/blob/master/docs/api.md#inject-todos-and-all-action-creators-addtodo-completetodo--as-actions

So you can later on use
```
 saveName() {
      return this.nameChoice === null ? 'unknown name' : this.props.saveName(this.nameChoice.value));
  }
````
And again it will be much simpler to split to container / component

where NameChoiceContainer will be simialr to
````
function mapStateToProps(state) {
  return { lastStep: state.nextStep }    //Do You need it ?
}

function mapDispatchToProps(dispatch) {
  return {
 saveName: (e) => dispatch(saveName(e.value))
}

export default connect(mapStateToProps, mapDispatchToProps)(NameChoiceComponent)
```

and NameChoiceComponent
```
export const NameChoiceComponent = ({lastStep, saveName}) => (
 <div>
          <input ref={(input) => { this.nameChoice = input; }} id="nameChoice" type="text" />
          <button onClick={saveName}>Set name</button>
  </div>
) 
```